Drop pyhdf

chuckwondo opened the pull request.

chuckwondo committed:
Make .gitignore more comprehensive

chuckwondo committed:
Shrink Docker image

Also, add Makefile to make it easy to build Docker image and run
Docker container

chuckwondo committed:
Parametrize test

chuckwondo committed:
Replace Makefile with compose file

chuckwondo committed:
Drop use of pyhdf

Fixes https://github.com/NASA-IMPACT/hls_development/issues/190

sharkinsspatial commented on Dockerfile:
@chuckwondo We should no longer need a full `osgeo/gdal:ubuntu` image for testing as we no longer need the HDF dependencies.  You should be able to use an alpine image with the plain rasterio 1.1.3 wheel installed.

chuckwondo replied:
Ah, cool. I'll update.

chuckwondo replied:
Do you have a specific image in mind?

I tried `remotepixel/alpine-rasterio:gdal3.0`, and I got that to work, but it has python 3.8 installed, not python 3.6, so I had to tweak `tox.ini` to use `envlist = py3` instead of `envlist = py36`. Is that what you're looking for?

chuckwondo replied:
I noticed that this image has rasterio 1.1.2 installed, so I explicitly pip installed 1.1.3, but that causes an exception:

```plain
 E   ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 96 from C header, got 80 from PyObject
```

chuckwondo replied:
@sharkinsspatial, I was spending too much time trying to wrestle with alpine images, so I simply changed from ubuntu-full to ubuntu-small and trimmed some system packages, which shaved about 1 GB from the image size. Is that sufficient?

sharkinsspatial commented on tests/test_thumbnails.py:
Can we download the thumbnail produced for the test granule by the original code and perform a checksum analysis to verify that our new output is the same?  I'm not sure if they will be identical given our methodology change but ideally we want the thumbnail to be almost identical.

chuckwondo replied:
Sure! I did manually test that the checksums are indeed identical. I will implement your suggestion to integrate checksum comparison directly in the unit tests.

chuckwondo replied:
Done

chuckwondo committed:
Improve unit tests by comparing checksums

chuckwondo committed:
Reduce Docker image size

Changed from "full" to "small" image and trimmed package installations,
reducing image size by ~1 GB.
